build/wd_compile_cache.bzl

```diff
@@ -0,0 +1,55 @@
+"""
+Bazel rules for generating compilation caches from a list of input files.
+
+This module provides rules and macros for creating compilation caches, which can be used
+to improve build performance by caching compilation artifacts.
+"""
+
+load("@bazel_skylib//rules:run_binary.bzl", "run_binary")
+
+def _gen_compile_cache_impl(ctx):
+    """Implementation of the gen_compile_cache rule.
+
+    This function:
+    1. Creates a file containing the list of input files
+    2. Runs a binary to generate the compilation cache
+    """
+    ctx.actions.write(
+        output = ctx.attr.file_list_path,
+        content = "\n".join(ctx.attr.srcs) + "\n",
+    )
+    run_binary(
+        ctx.attr.name,
+        srcs = ctx.attr.srcs,
+        outs = [ctx.attr.output],
+        args = [ctx.attr.file_list_path],
+        tool = ctx.attr._tool,
+    )
+    return compile_cache_output
+
+gen_compile_cache = rule(
+    implementation = _gen_compile_cache_impl,
+    attrs = {
+        "name": attr.string(mandatory = True, doc = "name"),
+        "srcs": attr.string_list(mandatory = True, doc = "files to create compile cache for"),
+        "file_list_path": attr.string(mandatory = True, doc = "temporary path for the file list"),
```

Comment:
sorry for noise, but temp files don't have to be here. I missed the fact that this list is not used anywhere else. The output does need to be here.